range_pool.go

```diff
@@ -190,6 +196,30 @@ func (s *Service) Delete(ctx context.Context, namespace, ID string) error {
 	return nil
 }
 
+func (s *Service) Search(ctx context.Context, namespace, ID string) ([]int, error) {
+	var used []int
+	{
+		k, err := microstorage.NewK(fmt.Sprintf(ItemSearchKeyFormat, namespace, ID))
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		kv, err := s.storage.List(ctx, k)
+		if microstorage.IsNotFound(err) || len(kv) == 0 {
```

Comment:
I am not sure how the behaviour should be when searching/listing here. I think returning an error in case what we want is not there is reasonable.

Reply:
Don't know enough about RangePool to help here. But its good that a not found error is returned. I'd say go with this.

Reply:
microstorage.List should never return notFoundError.